spec/windowsTitleBar.spec.ts

```diff
@@ -0,0 +1,195 @@
+        expect(spy).lastCalledWith(expect.any(Function));
+    });
+
+    describe('componentDidMount event', () => {
+        beforeEach(() => {
+            document.body.innerHTML = `<div id="content-wrapper"></div>`;
+        });
+
+        it('should call `componentDidMount` when isFullScreen', () => {
+            const spy = jest.spyOn(document.body.style, 'removeProperty');
+            const expectedValue = 'margin-top';
+            // changing state before componentDidMount
+            jest.spyOn(remote, 'getCurrentWindow').mockImplementation(() => {
+                return {
+                    isFullScreen: jest.fn(() => {
+                        return true;
+                    }),
+                    isMaximized: jest.fn(() => {
+                        return false;
+                    }),
+                    on: jest.fn(),
+                    removeListener: jest.fn(),
+                    isDestroyed: jest.fn(() => {
+                        return false;
+                    }),
+                    close: jest.fn(),
+                    maximize: jest.fn(),
+                    minimize: jest.fn(),
+                    unmaximize: jest.fn(),
+                    setFullScreen: jest.fn(),
+                };
+            });
+            shallow(React.createElement(WindowsTitleBar));
+            expect(spy).toBeCalledWith(expectedValue);
+        });
+    });
+
+    describe('maximize functions', () => {
+        it('should call `unmaximize` correctly when is not full screen', () => {
+            const titleLabel = 'Maximize';
+            const unmaximizeFn = 'unmaximize';
+            const customSelector = `button.title-bar-button[title="${titleLabel}"]`;
+            const wrapper = shallow(React.createElement(WindowsTitleBar));
+            const instance: any = wrapper.instance();
+            const window = instance.window;
+            const spy = jest.spyOn(window, unmaximizeFn);
+            wrapper.setState({ isMaximized: true });
+            wrapper.find(customSelector).simulate('click');
+            expect(spy).toBeCalled();
+        });
+
+        it('should call `unmaximize` correctly when is full screen', () => {
+            const windowSpyFn = 'setFullScreen';
+            const titleLabel = 'Maximize';
+            const customSelector = `button.title-bar-button[title="${titleLabel}"]`;
+            const wrapper = shallow(React.createElement(WindowsTitleBar));
+            const instance: any = wrapper.instance();
+            const window = instance.window;
+            const spy = jest.spyOn(window, windowSpyFn);
+            window.isFullScreen = jest.fn(() => {
+                return true;
+            });
+            wrapper.setState({ isMaximized: true });
+            wrapper.find(customSelector).simulate('click');
+            expect(spy).toBeCalledWith(false);
+        });
+
+        it('should call `maximize` correctly when is not full screen', () => {
```

Comment:
typo. please change statement to `should call maximize correctly when it is not in full screen`